glotaran/builtin/megacomplexes/pfid/pfid_megacomplex.py

```diff
@@ -0,0 +1,273 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+import xarray as xr
+from scipy.special import erf
+
+from glotaran.builtin.megacomplexes.decay.irf import Irf
+from glotaran.builtin.megacomplexes.decay.irf import IrfMultiGaussian
+from glotaran.model import DatasetModel
+from glotaran.model import ItemIssue
+from glotaran.model import Megacomplex
+from glotaran.model import Model
+from glotaran.model import ModelItemType
+from glotaran.model import ParameterType
+from glotaran.model import attribute
+from glotaran.model import item
+from glotaran.model import megacomplex
+
+if TYPE_CHECKING:
+    from glotaran.parameter import Parameters
+    from glotaran.typing.types import ArrayLike
+
+
+class OscillationParameterIssue(ItemIssue):
+    def __init__(self, label: str, len_labels: int, len_frequencies: int, len_rates: int):
+        self._label = label
+        self._len_labels = len_labels
+        self._len_frequencies = len_frequencies
+        self._len_rates = len_rates
+
+    def to_string(self) -> str:
+        return (
+            f"Size of labels ({self.len_labels}), frequencies ({self.len_frequencies}) "
+            f"and rates ({self.len_rates}) does not match for damped oscillation "
+            f"megacomplex '{self.label}'."
```

Comment:
This will crash with an `AttributeError` since the actual attributes have a leading `_` (inherited issue from DOAS Megacomplex).
Please add a test and fix.
I will create a PR for the DOAS.

Reply:
See #1513

Reply:
Also change `damped oscillation megacomplex` to `pfid`

Reply:
fixed in 2c65b49f6981b85069fd12952378f77ae0accad1